### ISSUE TITLE
Adding support for informational warnings about crates looking for new maintainers?

### ISSUE THREAD

**Issue by gendx** [RustSec/cargo-audit#247]
Currently, crates can be [flagged as unmaintained](https://docs.rs/rustsec/0.21.0/rustsec/advisory/informational/enum.Informational.html#variant.Unmaintained) in the advisory database, to warn about crates that are "unmaintained / abandoned".

However there can be many reasons for *reduced* maintenance of a crate.
- The crate is outright abandoned. *[covered now]*
- The crate is no longer maintained because a better replacement exists, e.g. with a stronger design, a larger community to support it. *[covered now]*
- The crate cannot currently be fully maintained due to a lack of support/resources. For example, pull requests have been merged since the last release, but there are pending issues preventing a new release. As a result, no new release has been published for a while, but with the help of more maintainers, the crate could be revived. Also, in that case there isn't necessarily a better replacement, and forking to a new crate is not necessarily desirable either.

The third case is not covered by the current warning. Indeed, such a crate is not "abandoned", nor really "unmaintained", but rather is in "reduced maintenance mode" (if that makes sense). Flagging such a crate as "abandoned/unmaintained" could be disrespectful to the contributors, and wouldn't convey all the nuance of the limited maintenance to those who depend on that crate.

Setting up a *temporary* warning (e.g. a warning valid only for versions up to the current one, so that newer releases of the crate "unlock" the maintenance status) on the advisory database could help raise awareness about the maintenance status of a crate, by having all users running `cargo audit` becoming aware of it, and potentially realizing that they could/should help maintain that crate, given that they rely on it.

I don't have a specific example in mind, but would such messages (maybe at a lower "info" level rather than "warning") be helpful on the advisory database? Or would they on the contrary be detrimental?

I could see the following potential drawbacks.
- Too much noise. In general, too much noise about security advisories can encourage people to simply ignore the advisories. However, clearly identifying the advisory levels as error/warning/info could help reduce this noise effect.
- Generally encouraging more frequent hand overs of crates to untrusted third-parties, which could increase the risk that a crate is handed over to a malicious party (the ["event-stream" problem](https://github.com/dominictarr/event-stream/issues/116))? However, helping maintain a crate doesn't necessarily mean becoming an official maintainer with publish rights on crates.io, it could simply be proposing pull requests to fix pending issues, so that a new release can happen.

**Comment by tarcieri** [comment]
I agree it'd might be good to clarify "looking for maintainer" versus "unmaintained". That's the state of the `term` crate, for example:

https://rustsec.org/advisories/RUSTSEC-2018-0015.html
